ipykernel/tests/utils.py

```diff
@@ -36,7 +36,7 @@ def start_new_kernel(**kwargs):
 
     Integrates with our output capturing for tests.
     """
-    stdout = getattr(nose, 'iptest_stdstreams_fileno', open(os.devnull))
+    stdout = getattr(nose, 'iptest_stdstreams_fileno()', open(os.devnull))
```

Comment:
This isn't doing quite what you expect. `getattr()` only looks for a named attribute, and because there can't be brackets in a name, it fails to find it and falls back to the default from the third argument.

I think you have the right idea to fix it properly. But I'm curious about what you're running to get this error - this test code has now been separated from the `iptest` runner that sets up `iptest_stdstreams_fileno`.